lib/Doctrine/DBAL/Driver/Mysqli/MysqliStatement.php

```diff
@@ -228,6 +245,63 @@ public function execute($params = null)
         return true;
     }
 
+    /**
+     * Move all stream resources that are bound with the LARGE_OBJECT type
+     * from the _bindesValues array over to longData so that we can
+     * send them using send_long_data.
+     */
+    private function prepareLongData()
+    {
+        foreach ($this->_bindedValues as $parameter => $value) {
+            if ($this->types[$parameter - 1] !== static::$_paramTypeMap[ParameterType::LARGE_OBJECT]) {
+                continue;
+            }
+
+            if (is_resource($value)) {
+                if (get_resource_type($value) !== 'stream') {
+                    throw new InvalidArgumentException('Resources passed with the LARGE_OBJECT parameter type must be stream resources.');
+                }
+                $this->_bindedValues[$parameter] = null;
+                $this->longData[$parameter - 1]  = $value;
+            } else {
+                $this->types[$parameter - 1] = static::$_paramTypeMap[ParameterType::STRING];
+            }
+        }
+    }
+
+    /**
+     * Handle $this->_longData after regular query parameters have been bound
+     *
+     * @throws MysqliException
+     */
+    private function processLongData()
+    {
+        foreach ($this->longData as $paramNr => $stream) {
+            while (! feof($stream)) {
+                $chunk = fread($stream, 8192);
+                if ($chunk === false) {
+                    throw new MysqliException("Failed reading the stream resource for parameter offset ${paramNr}.");
+                }
+                $this->sendLongData($paramNr, $chunk);
+            }
+        }
+    }
+
+    /**
+     * Bind parameters using send_long_data
+     *
+     * @param int    $paramNr  Parameter offset
+     * @param string $longData A chunk of data to send
+     *
+     * @throws MysqliException
+     */
+    private function sendLongData($paramNr, $longData) : void
+    {
+        if (! $this->_stmt->send_long_data($paramNr, $longData)) {
```

Comment:
Does this need to exist as a separate method? It's only called from `processLongData()`. It'd make sense to leave only one named `sendLongData()` since the "process" term is too abstract,